spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -72,16 +73,43 @@
 	 * Default {@link PubSubTemplate} constructor that uses {@link SimplePubSubMessageConverter}
 	 * to serialize and deserialize payloads.
 	 * @param publisherFactory the {@link com.google.cloud.pubsub.v1.Publisher} factory to
-	 * publish to topics
+	 * publish to topics. Only required if using publish methods
 	 * @param subscriberFactory the {@link com.google.cloud.pubsub.v1.Subscriber} factory
-	 * to subscribe to subscriptions
+	 * to subscribe to subscriptions. Only required if using subscriber or pull methods
 	 */
 	public PubSubTemplate(PublisherFactory publisherFactory,
 			SubscriberFactory subscriberFactory) {
 		this.publisherFactory = publisherFactory;
 		this.subscriberFactory = subscriberFactory;
-		this.subscriberStub = this.subscriberFactory.createSubscriberStub();
-		this.acknowledger = this.subscriberFactory.createAcknowledger();
+
+		if (this.subscriberFactory != null) {
+			this.subscriberStub = this.subscriberFactory.createSubscriberStub();
+			this.acknowledger = this.subscriberFactory.createAcknowledger();
+		}
+		else {
+			this.subscriberStub = null;
+			this.acknowledger = null;
+		}
+	}
+
+	/**
+	 * {@link PubSubTemplate} constructor that uses {@link SimplePubSubMessageConverter}
+	 * to serialize and deserialize payloads.
+	 * @param publisherFactory the {@link com.google.cloud.pubsub.v1.Publisher} factory to
+	 * publish to topics. Subscribe and pull methods are not supported
+	 */
+	public PubSubTemplate(PublisherFactory publisherFactory) {
+		this(publisherFactory, null);
+	}
+
+	/**
+	 * {@link PubSubTemplate} constructor that uses {@link SimplePubSubMessageConverter}
+	 * to serialize and deserialize payloads.
+	 * @param subscriberFactory the {@link com.google.cloud.pubsub.v1.Subscriber} factory
+	 * to subscribe to subscriptions. Publish methods are not supported
+	 */
+	public PubSubTemplate(SubscriberFactory subscriberFactory) {
+		this(null, subscriberFactory);
```

Comment:
With this two `null`-based constructor you have missed the fact to `Assert.notNull()` factories in the target method.
Otherwise, when I don provide a `PublisherFactory`, my first `PubSubTemplate.publsih()` will fail in the `public ListenableFuture<String> publish(final String topic, PubsubMessage pubsubMessage) {`, because of NPE.
That's not good and so informative: we need to be proactive and fail with the reasonable message using particular Spring utils.

I don't tell that you need new unit-tests, but with them you would get an issue early and proceed respectively.

Thanks for understanding!

P.S. We may come back to our rule "always unit tests in PRs!"